### PR TITLE
fix(types): match types with API response.

### PR DESCRIPTION
**Pull request by NicolasMontone**
# What?

Match types with API response.

# Why?

Related task: Related task: https://pluggy.atlassian.net/browse/INT-229

### CODE REVIEW

File: src/transforms.ts [hunk 1]
```diff
@@ -31,6 +31,8 @@ export function transformIdentity(identity: DeserializedIdentityResponse): Ident
   return {
     ...identity,
     birthDate: identity.birthDate && new Date(identity.birthDate),
+    createdAt: new Date(identity.createdAt),
```
Comment (JpOttaviano) [inline src/transforms.ts:34]: Do you we need to inform created and updated date params for entities like identity in the sdk? Shouldnt this be related only in the execution or the item?

**Review by JpOttaviano — APPROVED** [review]
Approved, with just the single question.

File: src/transforms.ts (hunk 1 above)
Comment (tmilar) [reply src/transforms.ts:34]: not sure if this is needed (probably not) but we are currently in the task of having the SDKs reflect the same behaviour as the API responses. If the value here is incorrect, then we should consider actually removing it from the API response itself first. I don't think there is enough merit to remove it though.
@Gabrielpanga what do you think?

File: src/types/identity.ts [hunk 2]
```diff
@@ -23,6 +25,10 @@ export type IdentityResponse = {
   addresses?: Address[]
   /** List of associated personal relationships */
   relations?: IdentityRelation[]
+  /** Date of the first connection */
+  createdAt: Date
+  /** Last connection sync date with the institution. */
```
Comment (tmilar) [inline src/types/identity.ts:30]: these comments don't seem to be accurate. We are referring to `IdentityResponse` here, not to "the connection"

File: src/types/investment.ts [hunk 3]
```diff
@@ -119,27 +119,27 @@ export type InvestmentRateType = typeof INVESTMENT_RATE_TYPES[number]
...
+  clearingFee?: number | null
   /** Fees charged by BM&F Bovespa as a source of operating income  */
-  stockExchangeFee?: number
+  stockExchangeFee?: number | null
   /** Fee by brokers to keep recordsin their home broker systems or on the trading desk */
-  custodyFee?: number
+  custodyFee?: number | null
   /** Amount paid to the Operator for the intermediation service  */
-  operatingFee?: number
+  operatingFee?: number | null
```
Comment (tmilar) [inline src/types/investment.ts:142]: are all these fields correct as both optional _and_ nullable? (`| null`)

Comment (NicolasMontone) [reply src/types/investment.ts:142]: No, just the `Identity` object is _optional_ and  _nullable_, my bad 👍

Comment (NicolasMontone) [reply src/types/investment.ts:142]: This matches with the API response

File: src/transforms.ts (hunk 1 above)
Comment (NicolasMontone) [reply src/transforms.ts:34]: Agree with Tomi's comment. I understand the point of not having it, but for now, the idea is to reflect the API response

File: src/types/identity.ts (hunk 2 above)
Comment (NicolasMontone) [reply src/types/identity.ts]: Thanks! You are right!

File: src/types/investment.ts (hunk 3 above)
Comment (NicolasMontone) [reply src/types/investment.ts:142]: Fixed! Now it's just _nullable_

**Pull request merged**